src/test/java/org/jenkinsci/plugins/workflow/steps/CatchErrorStepTest.java

```diff
@@ -110,6 +111,7 @@
         r.assertLogContains("execution continued", b);
     }
 
+    @Ignore("This fails with recent Jenkins versions; not sure if it should still be expected to work.")
```

Comment:
Hmm, I would've expected it to work, but I just checked and the test seems to have been passing for the wrong reason previously (the message "caught error" was part of an unexpected exception...), so it was actually already broken in the ways that matter. I am surprised I haven't seen any reports from users about this issue. This test being broken means that the step lost backwards compatibility in version 2.16 of this plugin for anyone using this syntax (we could probably fix it with a custom override of `StepDescriptor.newInstance`, but I'm not sure if it really matters, and that would be something for a different PR anyway). 

My best guess as to why the test is failing after the core update is something related to Pipeline logging and build completion, perhaps updating the parent POM would also fix the issue by pulling in test harness changes like https://github.com/jenkinsci/jenkins-test-harness/pull/210.

Reply:
After updating to the 4.1 parent, I'll retry the test; if that makes it pass, I'll drop the `@Ignore`. Otherwise I'll leave it to prevent blocking builds of this PR, and a separate PR will be needed to either drop the test or fix the "pass the message as only positional param" feature.

Reply:
With the 4.1 parent this indeed passes on my end, probably because the
```
java.lang.IllegalArgumentException: Expected named arguments but got caught error
```
now ends up in the log again (and that matches the `caught error` string check). So I dropped the `@Ignore`.